packages/query/src/query.ts

```diff
@@ -0,0 +1,178 @@
+import type * as UCAN from '@ipld/dag-ucan'
+import type { Link, ByteView, DID } from '@ipld/dag-ucan'
+import type { sha256 } from 'multiformats/hashes/sha2'
+
+// As per https://hackmd.io/YNH1W7XkT9ey90cLX3dF9A#Shards
+export interface Shard {
+  blocks: ByteView<unknown>[]
+  roots?: Set<Link<unknown>>
+}
+
+export type CARLink = Link<Shard, 1, 0x0202, typeof sha256.code>
+
+export declare function serve(context: Context): StoreService
+
+export interface StoreService {
+  /**
+   * Handles `AddInput` request. If DID in `with` field is not associated with
+   * any account responds with `AddError`. If service already has CAR with the
+   * given `link` responds with `AddOk`, otherwise responds with `AddPending`
+   * providing `.to` URL where client can upload desired CAR file.
+   *
+   * @param input
+   */
+  add(input: AddInput): Promise<AddResult>
+
+  /**
+   * Removes give link from the given `DID` group. If DID does not has such
+   * link `RemoveError` is returned. If account responsible for `DID` has no
+   * other `DID`s in which given link is member of user quota is adjusted, by
+   * freeing corresponding space.
+   */
+  remove(input: RemoveInput): Promise<RemoveResult>
+}
+
+export interface Context {
```

Comment:
@the-simian I think this is kind of bindings I'd need on the backend to implement the protocol client & server